detox/src/devices/drivers/android/emulator/EmulatorDriver.js

```diff
@@ -51,7 +46,7 @@ class EmulatorDriver extends AndroidDriver {
     await this._avdValidator.validate(avdName);
     await this._fixEmulatorConfigIniSkinNameIfNeeded(avdName);
 
-    const adbName = await this._allocateDevice(avdName);
+    const adbName = await this.allocateDevice(avdName) || this._createDevice();
```

Comment:
I have reasons to believe that you mistyped something here, probably you meant:

```js
const adbName = await this.allocateDevice(avdName) || await this._createDevice();
```

because otherwise you risk writing a Promise inside `adbName` variable.

<img width="912" alt="image" src="https://user-images.githubusercontent.com/1962469/88305094-c31fca00-cd11-11ea-92ff-d821ada871b1.png">

Reply:
Fixed